spores-pickling/src/test/scala/scala/spores/run/pickling/Pickling.scala

```diff
@@ -14,102 +14,75 @@ import SporePickler._
-
-    builder.hintTag(implicitly[FastTypeTag[Spore[Int, String]]])
-    pickler.pickle(s, builder)
-    val res = builder.result()
-
-    assert(res.value.toString.endsWith("""
-  "captured": 10
-}"""))
-
-    val reader = format.createReader(res.asInstanceOf[format.PickleType])
-    val up = pickler.unpickle("scala.spores.Spore[Int, String]", reader)
-    val us = up.asInstanceOf[Spore[Int, String]]
-    val res2 = us(5)
-    assert(res2 == "arg: 5, c1: 10")
-  }
+  implicit val staticOnly = static.StaticOnly
 
   @Test
-  def `simplified spore pickling`(): Unit = {
+  def `pickle/unpickle to/from JSON with one captured variable`(): Unit = {
     val v1 = 10
     val s = spore {
       val c1 = v1
       (x: Int) => s"arg: $x, c1: $c1"
     }
 
-    val pickler = SporePickler.genSporePickler[Int, String, Int]
-    val builder = implicitly[PickleFormat].createBuilder
-    val res = {
-      builder.hintTag(implicitly[FastTypeTag[Spore[Int, String]]])
-      pickler.pickle(s, builder)
-      builder.result()
-    }
-
+    val res = s.pickle
     assert(res.value.toString.endsWith("""
       |  "captured": 10
       |}""".stripMargin))
+
+    val up = res.value.unpickle[Spore[Int, String]]
+    val res2 = up(5)
+    assert(res2 == "arg: 5, c1: 10")
   }
 
   @Test
   def `pickling spore with two captured variables`(): Unit = {
     val v1 = 10
-    val v2 = "hello"
+    val v2 = "hello1"
     val s = spore {
       val c1 = v1
       val c2 = v2
       (x: Int) => s"arg: $x, c1: $c1, c2: $c2"
     }
 
-    val pickler = SporePickler.genSporeCMPickler[Int, String, (Int, String)]
+    val pickler = SporePickler.genSporePicklerUnpickler[Int, String, (Int, String)]
+  
     val format  = implicitly[PickleFormat]
     val builder = format.createBuilder
     val res = {
-      builder.hintTag(implicitly[FastTypeTag[Spore[Int, String]]])
       pickler.pickle(s, builder)
       builder.result()
     }
 
+    System.out.println(s"res1: ${res.value}")
     assert(res.value.toString.endsWith(""""captured": {
-      |    "$type": "scala.Tuple2[scala.Int,java.lang.String]",
       |    "_1": 10,
-      |    "_2": "hello"
+      |    "_2": "hello1"
       |  }
       |}""".stripMargin))
 
     val reader = format.createReader(res.asInstanceOf[format.PickleType])
-    val up = pickler.unpickle("scala.spores.Spore[Int, String]", reader)
-    val us = up.asInstanceOf[Spore[Int, String]]
-    val res2 = us(5)
-    assert(res2 == "arg: 5, c1: 10, c2: hello")
+    val unpickler = SporePickler.genSporeUnpickler[Int, String]
+    val up = unpickler.unpickle("", reader).asInstanceOf[Spore[Int, String]]
+    val res2 = up(5)
+    System.out.println(s"res2: ${res2.value}")
+    assert(res2 == "arg: 5, c1: 10, c2: hello1")
   }
 
   @Test
   def `simple pickling of spore with two captured variables`(): Unit = {
     val v1 = 10
-    val v2 = "hello"
-    val s = spore {
+    val v2 = "hello2"
+    val s: SporeWithEnv[Int, String] {type Captured = (Int, String)}= spore {
```

Comment:
Again, this is a different test from before, since the type of the spore is no longer inferred.